ui/browsing/genrepage.go

```diff
@@ -26,8 +28,23 @@ func NewGenrePage(genre string, pool *util.WidgetPool, contr *controller.Control
 
 func (g *genrePageAdapter) Title() string { return g.genre }
 
-func (g *genrePageAdapter) Filter() *mediaprovider.AlbumFilter {
-	return &mediaprovider.AlbumFilter{Genres: []string{g.genre}}
+func (g *genrePageAdapter) Filter() mediaprovider.AlbumFilter {
```

Comment:
Great catch. I found the root cause being the media providers (Jellyfin/Subsonic) updating the filter's attributes directly, on their `IterateAlbums` method.

They zero out the filters they can successfully translate to API parameters, so no further filtering is needed in-app. However, I think moving away from that pattern and avoiding these implementations from editing the received filter makes the code simpler to maintain. The approach I found to work is for these methods to require cloning the filter and making changes to that new instance instead of the original one.